flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -844,16 +843,19 @@ private Operation convertShowViews(SqlShowViews sqlShowViews) {
         return new ShowViewsOperation();
     }
 
-    /** Convert EXPLAIN statement. */
-    private Operation convertExplain(SqlExplain sqlExplain) {
-        Operation operation = convertSqlQuery(sqlExplain.getExplicandum());
-
-        if (sqlExplain.getDetailLevel() != SqlExplainLevel.EXPPLAN_ATTRIBUTES
-                || sqlExplain.getDepth() != SqlExplain.Depth.PHYSICAL
-                || sqlExplain.getFormat() != SqlExplainFormat.TEXT) {
-            throw new TableException("Only default behavior is supported now, EXPLAIN PLAN FOR xx");
+    /** Convert RICH EXPLAIN statement. */
+    private Operation convertRichExplain(SqlRichExplain sqlExplain) {
+        Operation operation;
+        SqlNode sqlNode = sqlExplain.getStatement();
+        if (sqlNode instanceof RichSqlInsert) {
+            operation = convertSqlInsert((RichSqlInsert) sqlNode);
+        } else if (sqlNode instanceof SqlSelect) {
+            operation = convertSqlQuery(sqlExplain.getStatement());
```

Comment:
Can't `sqlNode` be passed as a parameter?

Reply:
Flink has its own logic to validate the INSERT statement, which is different from calcite. Here we only validate the query part of the INSERT statement and check whether the sink schema is as same as the query schema if the statement is the INSERT statement.

Reply:
Do you mean that the results of executions 1 and 2 below are different?
```
// 1
convertSqlQuery(sqlExplain.getStatement());
// 2
convertSqlQuery(sqlNode);
```

Reply:
I mean using `SqlValidatorImpl#validate(insertStatement)` may have the different plans comparing to using `SqlToOperationConverter#convertSqlInsert`. The validation about query is the same...